--- a/eternal-darkness-decomp/src/game/game_fn_801E3388.c
+++ b/eternal-darkness-decomp/src/game/game_fn_801E3388.c
@@ -42,7 +42,7 @@
 extern void fn_801DA890(int);
 extern u32 fn_801E2E1C(u32, int, int);
 extern void* fn_801E3020(u32, u32, void*, int, int, void**, void***, void***, void**, u32);
-extern void* fn_800CD568(int, int, u32, void*, void*, u32, float);
+extern void* fn_800CD568(int, int, u32, void*, void*, u32, double);
 extern int fn_801CEB2C(u32);
 extern void* fn_801DA3B0(u32, u32, void*, u8, int, int, void**, void**, void**, void***);
 extern void* fn_801DA058(u32);
