--- a/eternal-darkness-decomp/src/game/game_fn_801E3388.c
+++ b/eternal-darkness-decomp/src/game/game_fn_801E3388.c
@@ -97,7 +97,7 @@
             *output_b = fn_80201B54(fn_800CD568(object->source,
                                                 (int)owner, object->flags,
                                                 output_a, output_d,
-                                                resource, lbl_80651248));
+                                                resource, 0.0f));
             *output_c = *output_b;
             if ((object->flags & 0xF) == 8 && (int)resource != 0) {
                 void* second_b;
