--- a/eternal-darkness-decomp/src/game/game_fn_801EFA68.c
+++ b/eternal-darkness-decomp/src/game/game_fn_801EFA68.c
@@ -48,7 +48,7 @@
 extern void OSInit(void);
 extern void DVDInit(void);
 extern void OSSetErrorHandler(int, void*);
-extern void fn_8015DAB0(char*, int, u32);
+extern void fn_8015DAB0(void*);
 extern void fn_80215D2C(int);
 extern void fn_801EF788(void);
 extern void fn_80216EAC(void);
@@ -89,7 +89,7 @@
 extern void fn_80145004(void);
 extern void fn_801EF400(void);
 
-void fn_801EFA68(DisplayConfig* config, int mode)
+void fn_801EFA68(DisplayConfig* config)
 {
     void* arenaLo;
     void* arenaHi;
@@ -104,7 +104,7 @@
     U16ToDouble heightValue;
     U16ToDouble widthValue;
 
-    fn_8015DAB0(label, mode, initialValue);
+    fn_8015DAB0(label);
     OSInit();
     DVDInit();
     fn_80215D2C(1);
@@ -123,12 +123,12 @@
     arenaLo = OSGetArenaLo();
     arenaHi = OSGetArenaHi();
     memset(arenaLo, 0, (u8*)arenaHi - (u8*)arenaLo);
+    width = lbl_8064C38C->width;
     height = lbl_8064C38C->height;
-    width = lbl_8064C38C->width;
     state->field_0 = lbl_802662C0;
     state->field_4 = (void*)(((u32)arenaLo + 31) & ~31);
     OSSetArenaLo((void*)(((u32)state->field_4 +
-        ((((width + 15) & ~15) * height) << 1) + 31) & ~31));
+        (((u16)((width + 15) & ~15) * height) << 1) + 31) & ~31));
 
     newLo = OSGetArenaLo();
     arenaHi = OSGetArenaHi();
