--- a/eternal-darkness-decomp/src/game/game_fn_8014B0F0.c
+++ b/eternal-darkness-decomp/src/game/game_fn_8014B0F0.c
@@ -38,12 +38,15 @@
 {
     int count;
     Set* set;
-    u16 bit;
     int mode;
     int i;
     Set* entry;
+    u16 bit;
     Set* right_set;
     int update;
+    u16 other_bit;
+    float *x, *y, *z;
+    void* context;
     Value value;
 
     if (left == 0) {
@@ -70,6 +73,7 @@
     entry = set;
     bit = 1;
     for (i = 0; i < count; i++) {
+            bit = (u16)bit;
         if ((set->active & bit) != 0 && entry->object88 != 0) {
             if (mode != 0) {
                 fn_8017FF24(entry->object88, 0, 8);
@@ -77,33 +81,35 @@
                 fn_8017FF24(entry->object88, 8, 0);
             }
         }
-        bit <<= 1;
+        bit = (u16)(bit << 1);
         entry = (Set*)((char*)entry + 4);
     }
 
     if (update == 0) {
         entry = set;
-        bit = 1;
+        other_bit = 1;
         for (i = 0; i < count; i++) {
-            if ((set->active & bit) != 0 && entry->object88 != 0) {
+            other_bit = (u16)other_bit;
+            if ((set->active & other_bit) != 0 && entry->object88 != 0) {
                 fn_8017FD6C(entry->object88);
-                set->active &= ~bit;
+                set->active &= ~other_bit;
             }
-            bit <<= 1;
+            other_bit = (u16)(other_bit << 1);
             entry = (Set*)((char*)entry + 4);
         }
         return;
     }
 
     {
-        void* context = fn_80201BC8(right_set);
-        float* x = &value.x;
-        float* y = &value.y;
-        float* z = &value.z;
+        x = &value.x;
+        y = &value.y;
+        z = &value.z;
+        context = fn_80201BC8(right_set);
         entry = set;
-        bit = 1;
+        other_bit = 1;
         for (i = 0; i < count; i++) {
-            if ((set->active & bit) != 0 && entry->object88 != 0) {
+            other_bit = (u16)other_bit;
+            if ((set->active & other_bit) != 0 && entry->object88 != 0) {
                 if (fn_8012FA54(context, entry->value48) != 0 ||
                     fn_8012DC94(context, entry->value48) == 0) {
                     short* position;
@@ -115,10 +121,10 @@
                     fn_8014B604(z, position + 2);
                 } else {
                     fn_8017FD6C(entry->object88);
-                    set->active &= ~bit;
+                    set->active &= ~other_bit;
                 }
             }
-            bit <<= 1;
+            other_bit = (u16)(other_bit << 1);
             entry = (Set*)((char*)entry + 4);
         }
     }
