--- a/eternal-darkness-decomp/src/game/game_fn_8014B0F0.c
+++ b/eternal-darkness-decomp/src/game/game_fn_8014B0F0.c
@@ -38,10 +38,14 @@
 {
     int count;
     Set* set;
-    u16 bit;
     int mode;
     int i;
     Set* entry;
+    u16 bit;
+    float* y;
+    int j;
+    float* x;
+    float* z;
     Set* right_set;
     int update;
     Value value;
@@ -70,6 +74,7 @@
     entry = set;
     bit = 1;
     for (i = 0; i < count; i++) {
+            bit = (u16)bit;
         if ((set->active & bit) != 0 && entry->object88 != 0) {
             if (mode != 0) {
                 fn_8017FF24(entry->object88, 0, 8);
@@ -77,19 +82,20 @@
                 fn_8017FF24(entry->object88, 8, 0);
             }
         }
-        bit <<= 1;
+        bit = (u16)(bit << 1);
         entry = (Set*)((char*)entry + 4);
     }
 
     if (update == 0) {
         entry = set;
         bit = 1;
-        for (i = 0; i < count; i++) {
+        for (j = 0; j < count; j++) {
+            bit = (u16)bit;
             if ((set->active & bit) != 0 && entry->object88 != 0) {
                 fn_8017FD6C(entry->object88);
                 set->active &= ~bit;
             }
-            bit <<= 1;
+            bit = (u16)(bit << 1);
             entry = (Set*)((char*)entry + 4);
         }
         return;
@@ -97,12 +103,13 @@
 
     {
         void* context = fn_80201BC8(right_set);
-        float* x = &value.x;
-        float* y = &value.y;
-        float* z = &value.z;
+        x = &value.x;
+        y = &value.y;
+        z = &value.z;
         entry = set;
         bit = 1;
-        for (i = 0; i < count; i++) {
+        for (j = 0; j < count; j++) {
+            bit = (u16)bit;
             if ((set->active & bit) != 0 && entry->object88 != 0) {
                 if (fn_8012FA54(context, entry->value48) != 0 ||
                     fn_8012DC94(context, entry->value48) == 0) {
@@ -118,7 +125,7 @@
                     set->active &= ~bit;
                 }
             }
-            bit <<= 1;
+            bit = (u16)(bit << 1);
             entry = (Set*)((char*)entry + 4);
         }
     }
