--- a/eternal-darkness-decomp/src/game/game_fn_8014ECD8.c
+++ b/eternal-darkness-decomp/src/game/game_fn_8014ECD8.c
@@ -58,8 +58,8 @@
     fn_80180DDC(entry, object_id);
 
     extent = *(volatile u16*)(object + 0x28);
+    *(u16*)(entry + 0x1C) = extent;
     scale = lbl_8065054C;
-    *(u16*)(entry + 0x1C) = extent;
     entry[0x14] = 6;
     *(u16*)(entry + 0x20) = 9;
     entry[0x18] &= ~0x20;
