--- a/src/game/game_fn_801CE384.c
+++ b/src/game/game_fn_801CE384.c
@@ -50,2 +50,2 @@
-void fn_801CE384(u32 type, const Vec3f* position, u16 value, u32 value24,
-                 u8 value1C, void** objects)
+void fn_801CE384(register u32 type, register const Vec3f* position, register u16 value, register u32 value24,
+                 register u8 value1C, register void** objects)
@@ -58,5 +58,5 @@
-    u32 object_type;
-    s16 count;
-    s16* index;
-    void** output;
-    int i;
+    register u32 object_type;
+    register s16 count;
+    register s16* index;
+    register void** output;
+    register int i;
